rfcs/0000-error-handling.md

```diff
@@ -0,0 +1,154 @@
+
+Errors are made more than just a string with the `Error` type.
+
+The complete indexing status API schema is:
+
+```graphql
+type Query {
+  indexingStatusForSubgraphName(subgraphName: String!): SubgraphIndexingStatus
+  indexingStatusesForSubgraphName(subgraphName: String!): [SubgraphIndexingStatus!]!
+  indexingStatuses(subgraphs: [String!]): [SubgraphIndexingStatus!]!
+}
+
+type SubgraphIndexingStatus {
+  subgraph: String!
+  synced: Boolean!
+  health: Health!
+  errors: [Error!]! # Sorted from first to last
+  chains: [ChainIndexingStatus!]!
+  node: String!
+}
+
+# Fields are moved from `EthereumIndexingStatus` to the interface.
+interface ChainIndexingStatus {
+  network: String!
+  chainHeadBlock: Block
+  earliestBlock: Block
+  latestBlock: Block
+  lastHealthyBlock: Block
+}
+
+type EthereumIndexingStatus implements ChainIndexingStatus { }
+
+# Renamed from `EthereumBlock` to `Block`
+type Block {
+  hash: Bytes!
+  number: BigInt!
+}
+
+type Error {
+	message: String!
+	
+	# Context for the error.
+	network: String!
+	block: Block!
+	handler: String
+}
+
+enum Health {
+  "Subgraph syncing normally"
+  Healthy,
+  "Subgraph syncing but with errors"
+  Unhealthy,
+  "Subgraph halted due to errors"
+  Failed,
+}
+```
+
+Since the indexing API is a view to the subgraph metadata, an `Error` type is also added to the metadata where subgraph errors are recorded. The `health` field is also added to the metadata, replacing the  `failed` flag which is deprecated.
+
+## Compatibility
+
+This does incompatible changes to the indexing status API, however that API was so far experimental and undocumented.
+
+The behavior of synced subgraphs when encountering an error is changed to keep indexing. Our consensus is that the current behavior is useful to nobody and this can only be an improvement.
+
+## Drawbacks and Risks
+
+- Changing the behavior of errors on synced subgraphs has risks. There may be unanticipated consequences which may slip through testing. We may have gotten our rationale wrong and users may find the new behavior worse for factors we did not consider.
+
+## Alternatives & Extensions
+
+- Don't change the default behavior, and put make the "continue on failure" mode opt-in through a flag on the manifest. With the possible intention of making it the default in the future. This exposes us to less risk when rolling this out. The downsides are that adoption and feedback will be slower and developers will need to update their subgraphs in order to benefit.
+- Future extension: Allow mappings to define a trap handler that catches traps on a specific trigger handler. The mapping can then decide whether the subgraph should fail, be set as unhealthy, or just continue normally.
+
+## Open Questions
+
+- The field name `lastHealthyBlock` is up for bikeshedding, aternatives: `failedAfter`, `blockBeforeFailure`, `lastGoodBlock`...
+- Do we want to keep the `node` field in the indexing status API? It feels too iternal.
```

Comment:
We added it for the indexer agent to be able to balance work across multiple index nodes. I think it's worth keeping.

Reply:
Alright lets keep it.